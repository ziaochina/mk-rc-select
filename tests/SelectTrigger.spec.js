/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import SelectTrigger from '../src/SelectTrigger';

describe('SelectTrigger', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <SelectTrigger
        prefixCls="rc-select"
        transitionName="slide"
      >
        <div>foo</div>
      </SelectTrigger>
    );

    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('set popupTransitionName if animation given', () => {
    const wrapper = mount(
      <SelectTrigger
        prefixCls="rc-select"
        animation="slide-up"
      >
        <div>foo</div>
      </SelectTrigger>
    );

    expect(wrapper.find('Trigger').props().popupTransitionName).toBe('rc-select-dropdown-slide-up');
  });
});
