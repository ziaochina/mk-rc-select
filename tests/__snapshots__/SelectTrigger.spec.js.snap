// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTrigger renders correctly 1`] = `
<SelectTrigger
  prefixCls="rc-select"
  transitionName="slide"
>
  <Trigger
    action={Array []}
    afterPopupVisibleChange={[Function]}
    blurDelay={0.15}
    builtinPlacements={
      Object {
        "bottomLeft": Object {
          "offset": Array [
            0,
            4,
          ],
          "overflow": Object {
            "adjustX": 0,
            "adjustY": 1,
          },
          "points": Array [
            "tl",
            "bl",
          ],
        },
        "topLeft": Object {
          "offset": Array [
            0,
            -4,
          ],
          "overflow": Object {
            "adjustX": 0,
            "adjustY": 1,
          },
          "points": Array [
            "bl",
            "tl",
          ],
        },
      }
    }
    defaultPopupVisible={false}
    destroyPopupOnHide={false}
    focusDelay={0}
    getDocument={[Function]}
    getPopupClassNameFromAlign={[Function]}
    hideAction={
      Array [
        "click",
      ]
    }
    mask={false}
    maskClosable={true}
    mouseEnterDelay={0}
    mouseLeaveDelay={0.1}
    onPopupAlign={[Function]}
    onPopupVisibleChange={[Function]}
    popup={
      <DropdownMenu
        defaultActiveFirstOption={undefined}
        dropdownMenuStyle={undefined}
        firstActiveValue={undefined}
        inputValue={undefined}
        menuItems={undefined}
        multiple={undefined}
        onMenuDeselect={undefined}
        onMenuSelect={undefined}
        onPopupFocus={undefined}
        prefixCls="rc-select-dropdown"
        value={undefined}
        visible={undefined}
      />
    }
    popupAlign={Object {}}
    popupClassName="rc-select-dropdown--single"
    popupPlacement="bottomLeft"
    popupStyle={Object {}}
    popupTransitionName="slide"
    prefixCls="rc-select-dropdown"
    showAction={
      Array [
        "click",
      ]
    }
    transitionName="slide"
  >
    <div
      id="TriggerContainer"
    >
      <div
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        foo
      </div>
    </div>
  </Trigger>
</SelectTrigger>
`;
